docs/rules/max-lines.md

```diff
@@ -0,0 +1,117 @@
+# enforce a maximum file length (max-lines)
+
+Some people consider large files a code smell. Large files tend to do a lot of things and can make following what's going on hard.
+
+## Rule Details
+
+This rule enforces a maximum number of lines per file, in order to aid in maintainability and reduce complexity.
+
+
+## Options
+
+This rule has a number or object option:
+
+* `"max"` (default `100`) enforces a maximum number of lines in a file
```

Comment:
Thanks @TimvdLippe good catch. I intended to leave it as it is in JSCS (100) when I noticed it was different but forgot to update the code.

@mysticatea why did you change it from 100 to 300? @eslint/eslint-team any opinions on what value this should be?